### PR TITLE
tests: drivers: rpc_flash for nrf5340

### PR DESCRIPTION
**Pull request by sigvartmh**
Flash driver test targeting specifically netcore of nrf53.

Redo of: 
https://github.com/nrfconnect/sdk-nrf/pull/14292

### CODE REVIEW

**Review by katgiadla — CHANGES_REQUESTED** [review]
Please specify the pearson responsible for test in file `CODEOWNERS`.

File: tests/drivers/rpc_flash/testcase.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+common:
+  tags:
+    - drivers
+    - flash
+tests:
+  drivers.rpc_flash:
+    platform_allow: nrf5340dk_nrf5340_cpuapp
```
Comment (katgiadla) [inline tests/drivers/rpc_flash/testcase.yaml:7]: incorrect platform name convention (currenlty the name should be compatibile with HWMv2)
```suggestion
    platform_allow: nrf5340dk/nrf5340/cpuapp
```

File: tests/drivers/rpc_flash/testcase.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+    integration_platforms:
+      - nrf5340dk_nrf5340_cpuapp
```
Comment (katgiadla) [inline tests/drivers/rpc_flash/testcase.yaml:9]: like above:
```suggestion
      - nrf5340dk/nrf5340/cpuapp
```

**Review by katgiadla — CHANGES_REQUESTED** [review]

File: tests/drivers/rpc_flash/testcase.yaml (hunk 1 above)
Comment (sigvartmh) [reply tests/drivers/rpc_flash/testcase.yaml]: Fixed. forgot to update to HWmod2

File: tests/drivers/rpc_flash/testcase.yaml (hunk 2 above)
Comment (sigvartmh) [reply tests/drivers/rpc_flash/testcase.yaml]: Fixed. forgot to update to HWmod2

**Review by michalek-no — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]

**Comment by de-nordic** [comment]
@katgiadla Can you re-review?

**Review by katgiadla — APPROVED** [review]

**Review by katgiadla — APPROVED** [review]

**Review by nvlsianpu — APPROVED** [review]

**Pull request merged**